app/helpers/sessions_helper.rb

```diff
@@ -18,11 +18,21 @@ def sign_out
     @current_user = nil
   end
 
-  def user_path
+  def user_root_path
+    return request.referer if current_user.blank?
+    
     if current_user.is_client?
       reservations_path
     elsif current_user.is_planner?
       reservation_frames_path
     end
   end
+
+  def users_path
+    "/#{current_user.type.downcase}s"
+  end
+
+  def user_path
+    "#{users_path}/#{current_user.id}"
```

Comment:
```suggestion
  def user_path(user_id)
    currrent_user.is_client? ? client_path : planner_path
```

でいけそう...（できなかったら教えて😂

Reply:
ありがとうございます